src/Connection.php

```diff
@@ -234,9 +234,17 @@ protected function getHostDsn(array $config): string
         $hosts = is_array($config['host']) ? $config['host'] : [$config['host']];
 
         foreach ($hosts as &$host) {
-            // Check if we need to add a port to the host
-            if (strpos($host, ':') === false && ! empty($config['port'])) {
-                $host = $host.':'.$config['port'];
+            // ipv6
+            if (filter_var($host, \FILTER_VALIDATE_IP, \FILTER_FLAG_IPV6)) {
+                $host = '['.$host.']';
```

Comment:
I had to check that libmongoc supports this notation, and it is handled in [`_mongoc_host_list_from_string_with_err`](https://github.com/mongodb/mongo-c-driver/blob/1.24.4/src/libmongoc/src/mongoc/mongoc-host-list.c#L221).

Reply:
I did test before doing the PR 😉